src/domain/consents/revoke.ts

```diff
@@ -0,0 +1,84 @@
+/* istanbul ignore file */
+
+/*
+ * This flag is to ignore BDD testing for model
+ * which will be addressed in the future in
+ * ticket #354
+ */
+
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop
+ files are distributed onan 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+ --------------
+ ******/
+
+import { Request } from '@hapi/hapi'
+import { consentDB } from '../../lib/db'
+import { Consent } from '../../model/consent'
+import { Enum } from '@mojaloop/central-services-shared'
+import Logger from '@mojaloop/central-services-logger'
+import SDKStandardComponents from '@mojaloop/sdk-standard-components'
+
+/**
+ * Validates if the request is initiated by a valid source
+ * Compares Consent object's initiator ID with source
+ */
+export function isConsentRequestInitiatedByValidSource (
```

Comment:
We should factor this out. Ask Lewis or @eoln  where it should live.

Reply:
I think `validators.ts` would be a good place to keep all validation related code and then use `validators.` prefix
```
import validators from './validators.ts'
.....
if( validators.isConsentRequestInitiatedByValidSource(.....) ) { .... }
```
so code will be more readable.... if the names of validators will be reasonably short ;-)

Reply:
makes sense - so `validators.ts` would live in domain folder? 
Was talking to @lewisdaly  about possible having it live in the `shared` or `lib` folder?

Reply:
I wouldn't keep them in `shared` which is dedicated to low-level tools. Let `validators.ts` live in `/src/domain/validators.ts`